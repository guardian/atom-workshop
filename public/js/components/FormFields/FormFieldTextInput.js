import React, {PropTypes} from 'react';
import ShowErrors from '../Utilities/ShowErrors';
import { errorPropType } from '../../constants/errorPropType';

export default class FormFieldTextInput extends React.Component {


  static propTypes = {
    fieldLabel: PropTypes.string,
    fieldName: PropTypes.string,
    fieldValue: PropTypes.string,
    fieldPlaceholder: PropTypes.string,
    fieldErrors: PropTypes.arrayOf(errorPropType),
    formRowClass: PropTypes.string,
    onUpdateField: PropTypes.func
  };

  onUpdate = (e) => {
    this.props.onUpdateField(e.target.value);
  }

  render() {
    return (
        <div className={this.props.formRowClass || "form__row"}>
          {this.props.fieldLabel ? <label htmlFor={this.props.fieldName} className="form__label">{this.props.fieldLabel}</label> : false}
          <input type="text" className={"form__field " + (this.props.fieldErrors && this.props.fieldErrors.length ? "form__field--error" : "")}  id={this.props.fieldName} placeholder={this.props.fieldPlaceholder || ''} onChange={this.onUpdate}  value={this.props.fieldValue || ""}/>
          <ShowErrors errors={this.props.fieldErrors}/>
        </div>

    );
  }
}
