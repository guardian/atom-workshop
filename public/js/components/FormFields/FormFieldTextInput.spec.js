import React from 'react';
import FormFieldTextInput from './FormFieldTextInput';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValue= 'test';

test('Should render', () => {

  const updateFn = jest.fn();
  const component = renderer.create(
    <FormFieldTextInput fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call update function on change', () => {

  const updateFn = jest.fn();
  const input = shallow(
    <FormFieldTextInput fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );

  input.find('input').simulate('change', {target: {value: "test"}});

  expect(updateFn).toHaveBeenCalledTimes(1);
});
