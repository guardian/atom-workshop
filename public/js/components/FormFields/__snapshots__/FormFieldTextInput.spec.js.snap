exports[`test Should render 1`] = `
<div>
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <input
    className="form__field "
    id="test"
    onChange={[Function]}
    placeholder=""
    type="text"
    value="test" />
  <div />
</div>
`;
